### DISCUSSION TITLE
Influence of font-weight seems not addressed

### DISCUSSION THREAD

**Discussion by andriytyurnikov** [seek-oss/capsize#36] [Q&A] [answered]
While it looks like results of capsize computations are legit for normal font-wight, it looks like font weight affects character proportions in special way, which means that we have to compute values for each font-weight.
<img width="185" alt="Screenshot 2021-06-08 at 12 38 08" src="https://user-images.githubusercontent.com/3668/121162963-11b95280-c857-11eb-98e0-11d3d31fc2d1.png">

From this perspective, option of optical computation on canvas may again become attractive.

**Comment by michaeltaranto** [comment, marked as answer]
Hey 👋 

Most fonts will have different metrics for the different weights, so you if you are after that precision for different weights you need to ensure you're using the correct metrics for the correct weight.

However, if you are using heavier text alongside normal text you may want not want that precision as the heavy text may appear too high.

In the case of artificial weight, where the browser has scaled the weight, you can adjust the font metrics manually to suit.

Capsize is not designed to be a runtime solution, so rendering to canvas to compute values is not feasible.

**Reply by andriytyurnikov** [reply]
Hi 👋

Well, sure, I see that every **_capsize_-weight** pair would need it's own font-size calculated for pixel-perfect alignment.

Honestly I am not aware of nuances of visual _perception_ of typography, but scenario I had in my mind is where bold and normal text have perfectly same capsize and/or baseline. By no means I am saying it is an only/necessary/worthy goal, but it felt like this is a right place to talk about pixel-perfect stuff 🤗

Subjectively it feels like current solution is best for occasional bold words, although I am not sure about titles, buttons, etc.

Well, capsize library is an awesome runtime solution, but capsize web page may be different 😅

I believe I've seen canvas-based calculating tool somewhere on the internets, so I created this ticket not for solution, but hoping to learn something from rationale behind current approach, as I am sure you are well aware of this nuanced case.

Who knows, maybe your rationale would be helpful to people in W3C 🧐
